src/js/sharedEth/utils.js

```diff
@@ -61,6 +61,26 @@ function providerType(provider) {
   }
 }
 
+function providerTypeId(provider) {
+  if (provider === undefined || provider == null) {
+    return 3;
+  } else if (provider.isToshi) {
+    return 3;
+  } else if (provider.isImToken) {
+    return 3;
+  }else if (provider.isTally) {
+    return 5;
+  } else if (provider.isMetaMask) {
+    if (provider.constructor.name === 'InpageBridge') {
+      return 3;
+    } else {
+      return 3;
```

Comment:
is this intentional to return 3 on both of these conditions? if so we can probably remove the check for `InpageBridge` above.